### PR TITLE
♻️ Move `context` under `#core`

### PR DESCRIPTION
**Pull request by rcebulko**
Part of #32693

### CODE REVIEW

**Review by samouri — APPROVED** [review]

File: src/core/context/node.js [hunk 1]
```diff
@@ -66,7 +66,7 @@ export class ContextNode {
     let contextNode = /** @type {!ContextNode|undefined} */ (node[NODE_PROP]);
     if (!contextNode) {
       contextNode = new ContextNode(node, null);
-      if (getMode().localDev || getMode().test) {
+      if (mode.isLocalDev() || mode.isTest()) {
```
Comment (micajuine-ho) [inline src/core/context/node.js:69]: For my reference, where are these methods defined?

**Review by micajuine-ho — APPROVED** [review]

File: src/core/context/node.js [hunk 2]
```diff
@@ -14,9 +14,9 @@
  * limitations under the License.
  */
 
+import * as mode from '#core/mode';
```
Comment (rcebulko) [inline src/core/context/node.js:17]: @micajuine-ho They're defined here :)

**Comment by rcebulko** [comment]
I suspect this may fit under `#core/dom`, but I wasn't sure, so pulled it into `#core` for now and can move it into `#core/dom` if that makes sense once @caroqliu is back/has a chance to look :)

**Pull request merged**